### DISCUSSION TITLE
allow to use remapped left / right mouse buttons. Make app work if you set RMB as LMB - for left-handed ppl

### DISCUSSION THREAD

**Discussion by seahorsee** [HerMajestyDrMona/Windows11DragAndDropToTaskbarFix#64] [Ideas]
Hi there,
The app worked great for me, but i have to use my mouse with a left hand for some time (broken arm). If you use a mouse with a left hand, the most comfortable setting is to change the default mouse button to right (RMB becomes LMB and vice versa), so you still click with your forefinger. When windows is set this way, the app doesn't work. I have looked over your configuration manual, however the key remapping option seems to target keyboard keys, not a mouse.
Would it be possible to add a parameter in configuration to swap LMB with RMB inside the app so it will still work? For ppl in a situation like mine and all left handed people this could prove handy.
Cheers!

**Comment by HerMajestyDrMona** [comment]
Hello. This is strange, because according to:
https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix#how-does-it-work

"The program detects if you're currently pressing the Left or the Right Mouse Button and determines which icon on the taskbar you hover the mouse pointer on."

So the program already detects the Right and the Left mouse click for the drag and drop gesture. Therefore there should be nothing more needed to fix, but maybe Windows does something extra? Hard to say. I'll test it and let you know.

**Comment by seahorsee** [comment]
thanks for quick reply. 
indeed there is something funky going on. If i use a defaul layout (LMB=LMB), your app works with both left and right mouse buttons. But when using switched layout, no mouse button seems to work. Tested with mouses and trackpads, the same story. For clarity i only use the default mouse settings in windows.

**Reply by HerMajestyDrMona** [reply]
Can you please try this test version and see if everything works fine?
You can download it here: https://we.tl/t-yy8Jlh36nV

There was indeed a bug where `GetAsyncKeyState(VK_LBUTTON)` functions (added in newer versions of the program which fix other bugs...) returned the hardware status of the mouse button, so the program was ignoring clicks when the Primary Mouse Button was set to the Right. In the newest version I added an additional check with `GetSystemMetrics(SM_SWAPBUTTON)`, so that the program will automatically switch to the Right primary mouse button mode and everything should work fine :)

**Comment by seahorsee** [comment]
This does fix it.
thanks a lot!

**Reply by HerMajestyDrMona** [reply]
That's great. Thanks for confirmation. I'll add it to the next version.

**Comment by music23** [comment]
I was looking to make my trackpad left handed as I am left handed and this is the way you can also [make your mouse/trackpad left handed in windows 11](https://thedigitalbin.com/how-to-make-the-mouse-or-trackpad-left-handed-in-windows-11/).

**Reply by HerMajestyDrMona** [reply]
Yes, the support for the Right Mouse Button as the primary one was added in [version 2.3.0.0](https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/blob/main/CHANGELOG.md#ver-2300----2022-02-15-1730-cet):

> - Fixed a bug where the program did not work when the Primary Mouse Button was changed to Right in Windows 11 Settings -> Bluetooth & Devices -> Mouse. Thanks for the report in discussion https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/discussions/64.
